pkg/storage/azure/azure.go

```diff
@@ -360,15 +355,33 @@ func (d *driver) StorageExists(cr *imageregistryv1.Config) (bool, error) {
 	_, err = container.GetProperties(d.Context, azblob.LeaseAccessConditions{})
 	if e, ok := err.(azblob.StorageError); ok {
 		if e.ServiceCode() == azblob.ServiceCodeContainerNotFound {
-			util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonContainerNotFound, "Container does not exist")
 			return false, nil
 		}
 	}
 	if err != nil {
-		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionUnknown, storageExistsReasonAzureError, fmt.Sprintf("Unable to get the storage container: %s", err))
 		return false, fmt.Errorf("unable to get the storage container %s: %s", d.Config.Container, err)
 	}
 
+	return true, nil
+}
+
+// StorageExists checks if the storage container exists and is accessible.
+func (d *driver) StorageExists(cr *imageregistryv1.Config) (bool, error) {
+	if d.Config.AccountName == "" || d.Config.Container == "" {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonNotConfigured, "Storage is not configured")
+		return false, nil
+	}
+
+	exists, err := d.containerExists(d.Config.Container)
+	if err != nil {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonAzureError, fmt.Sprintf("%s", err))
+		return false, err
+	}
+	if !exists {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonContainerNotFound, "")
```

Comment:
Provide a message if we couldn't find the storage container:
```suggestion
		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapiv1.ConditionFalse, storageExistsReasonContainerNotFound, fmt.Sprintf("Could not find storage container %s, d.Config.Container))
```

Reply:
updated in next push